app/scripts/lib/account-tracker.js

```diff
@@ -110,25 +111,147 @@ export default class AccountTracker {
         ) {
           this._updateAccounts();
         }
-      }, this.onboardingController.store.getState()),
+      }, this.preferencesController.store.getState()),
     );
-    this.ethersProvider = new Web3Provider(this._provider);
   }
 
+  /**
+   * Starts polling with global selected network
+   */
   start() {
     // remove first to avoid double add
     this._blockTracker.removeListener('latest', this._updateForBlock);
     // add listener
     this._blockTracker.addListener('latest', this._updateForBlock);
     // fetch account balances
-    this._updateAccounts();
+    this._updateAccountsWithNetworkClientId();
   }
 
+  /**
+   * Stops polling with global selected network
+   */
   stop() {
     // remove listener
     this._blockTracker.removeListener('latest', this._updateForBlock);
   }
 
+  /**
+   * Resolves a networkClientId to a network client config
+   * or globally selected network config if not provided
+   *
+   * @param networkClientId - Optional networkClientId to fetch a network client with
+   * @returns network client config
+   */
+  getCorrectNetworkClient(networkClientId) {
```

Comment:
Lets make this private
```suggestion
  _getCorrectNetworkClient(networkClientId) {
```

Reply:
done https://github.com/MetaMask/metamask-extension/pull/21868/commits/0455aca0a8fe4ac64823f6bff387cbdce7c56904

Reply:
done https://github.com/MetaMask/metamask-extension/pull/21868/commits/0455aca0a8fe4ac64823f6bff387cbdce7c56904